torchaudio/sox_effects/sox_effects.py

```diff
@@ -153,7 +155,7 @@ def apply_effects_tensor(
 
 @_mod_utils.requires_module('torchaudio._torchaudio')
 def apply_effects_file(
-        path: str,
+        path: Union[str, Path],
```

Comment:
You can remove the Union and just use `str`, but explaining why in the docstring which is how it is done in sox_io.

Reply:
I second this idea :)

Reply:
@vincentqb changed type as per sox_io.
Also used `str()` like sox_io to convert Path to string, as` os.fspath()` is not supported by torchscript.
Now all checks passed.